src/Node.php

```diff
@@ -347,22 +347,21 @@ public function getNodeKindName() : string {
      */
     public function getEndPosition() {
         // TODO test invariant - start of next node is end of previous node
-        if (isset($this->parent)) {
-            $parent = $this->parent;
-            $siblings = $parent->getChildNodes();
-            foreach ($siblings as $idx=>$nextSibling) {
-                if (spl_object_hash($nextSibling) === spl_object_hash($this)) {
-                    $siblings->next();
-                    $nextSibling = $siblings->current();
-                    return $nextSibling !== null
-                        ? $nextSibling->getFullStart()
-                        : $this->getRoot()->endOfFileToken->fullStart;
-                }
+        if (!isset($this->parent)) {
+            throw new \Exception("Unhandled node: " );
+        }
+
+        $parent = $this->parent;
+        $siblings = $parent->getChildNodes();
```

Comment:
not related to your change in particular, but reading through this made me realize that $parent isn't getting used anywhere else, so it makes sense to just remove that assignment, and use `$siblings = $this->parent->getChildNodes();`